firestore-translate-text/__tests__/functions.test.ts

```diff
@@ -0,0 +1,384 @@
+const mockTranslateClassMethod = jest
+  .fn()
+  .mockImplementation((string: string, targetLanguage: string) => {
+    return Promise.resolve([testTranslations[targetLanguage]]);
+  });
+
+// new Translate(opts);
+const mockTranslateClass = jest.fn().mockImplementation(() => {
+  return { translate: mockTranslateClassMethod };
+});
+
+// import { Translate } from "@google-cloud/translate";
+function mockTranslateModuleFactory() {
+  return {
+    Translate: mockTranslateClass,
+  };
+}
+
+function mockDocumentSnapshotFactory(documentSnapshot) {
+  return jest.fn().mockImplementation(() => {
+    return {
+      exists: true,
+      get: documentSnapshot.get.bind(documentSnapshot),
+      ref: {
+        path: documentSnapshot.ref.path,
+      },
+    };
+  })();
+}
+
+jest.mock("@google-cloud/translate", mockTranslateModuleFactory);
+
+let restoreEnv;
+let functionsTest = functionsTestInit();
+const mockConsoleLog = jest.spyOn(console, "log").mockImplementation();
+const mockConsoleError = jest.spyOn(console, "error").mockImplementation();
+
+describe("extension", () => {
+  beforeEach(() => {
+    restoreEnv = mockedEnv(defaultEnvironment);
+    mockConsoleLog.mockClear();
+    mockConsoleError.mockClear();
+  });
+
+  afterEach(() => restoreEnv());
+
+  test("functions are exported", () => {
+    const exportedFunctions = jest.requireActual("../functions/src");
+    expect(exportedFunctions.fstranslate).toBeInstanceOf(Function);
+  });
+
+  describe("functions.fstranslate", () => {
+    let admin;
+    let wrappedFunction;
+    let beforeSnapshot;
+    let afterSnapshot;
+    let documentChange;
+
+    beforeEach(() => {
+      jest.resetModules();
+      functionsTest = functionsTestInit();
+      admin = require("firebase-admin");
+      wrappedFunction = functionsTest.wrap(
+        require("../functions/src").fstranslate
+      );
+
+      beforeSnapshot = functionsTest.firestore.makeDocumentSnapshot(
+        {},
+        "translations/id1"
+      );
+      afterSnapshot = functionsTest.firestore.makeDocumentSnapshot(
+        { input: "hello" },
+        "translations/id1"
+      );
+      documentChange = functionsTest.makeChange(
+        beforeSnapshot,
+        mockDocumentSnapshotFactory(afterSnapshot)
+      );
+      admin.firestore().runTransaction = jest.fn().mockImplementation(() => {
+        return (transactionHandler) => {
+          transactionHandler({
+            update(ref, field, data) {
+              mockFirestoreUpdate(field, data);
+            },
+          });
+        };
+      })();
+    });
+
+    test("initializes Google Translate API with PROJECT_ID on function load", () => {
+      jest.resetModules();
+      mockTranslateClass.mockClear();
+
+      require("../functions/src");
+
+      expect(mockTranslateClass).toHaveBeenCalledTimes(1);
+      expect(mockTranslateClass).toHaveBeenCalledWith({
+        projectId: defaultEnvironment.PROJECT_ID,
+      });
+    });
```

Comment:
I was confused by this test. On the surface I don't see what the relationship is between the require and the assertions.  Perhaps add some comments or change the string to describe the expected side effect?